workflow/automation/execution_scripts/cybershake_progress.py

```diff
@@ -34,53 +35,48 @@
 SLACK_ALERT_CONFIG = "slack_alert.yaml"
 
 _fault_template_fmt = "{:<15}{:<22}{:<7}{:<3}{:<17}{:<7}{:<3}{:<17}{:<7}{:<3}{:<22}"
-_summary_template_fmt = "{:<6} : {:>5}/{:>5} ({:4.2f}%) RELs complete: {:10.2f} CH consumed / {:10.2f} CH est ({: >#06.2f}%) {:10.2f} CH remaining est"
-
-
-def get_chours(json_file, proc_types: List[str], debug=False):
-    """Gets the core hours for the specified process types (i.e. EMOD3D, HF, BB, ...)
-    for the specified simulation metadata json log file.
-    """
-    with open(json_file, "r") as f:
-        data = json.load(f)
-    sim_name = data.get(const.MetadataField.sim_name.value)
-    core_hours = []
-    completed_proc = {}
-    for proc_type in proc_types:
-        if proc_type in data.keys():
-            core_hours.append(
-                data[proc_type].get(const.MetadataField.run_time.value, np.nan)
-                * data[proc_type].get(const.MetadataField.n_cores.value, np.nan)
-                / 3600.0
-            )
+_summary_con_template_fmt = (
+    "{:<6} : {:>5}/{:>5} ({:4.2f}%) RELs complete: {:10.2f} CH consumed"
+)
+_summary_est_template_fmt = (
+    " / {:10.2f} CH est ({: >#06.2f}%) {:10.2f} CH remaining est"
+)
 
-        else:
-            core_hours.append(np.nan)
-    return sim_name, core_hours
 
-
-def get_chours_used(fault_dirs: Iterable[str]):
+def get_chours_used(root_dir: str, fault_names: List[str], proc_types: List[str]):
     """Returns a dataframe containing the core hours used for each of the
     specified faults"""
-    faults, core_hours, completed_r_counts, missing_data = [], [], [], []
-    for fault in fault_dirs:
-        fault = Path(fault)
-        fault_name = fault.stem
-        cur_fault_chours = np.asarray(
-            [
-                get_chours(json_file, PROCESS_TYPES)[1]
-                for json_file in fault.glob(f"**/{const.METADATA_LOG_FILENAME}")
-            ]
-        ).reshape(-1, len(PROCESS_TYPES))
-
-        faults.append(fault_name)
-        if cur_fault_chours.shape[0] > 0:
-            core_hours.append(np.nansum(cur_fault_chours, axis=0))
-        else:
-            core_hours.append([np.nan, np.nan, np.nan])
-
-    df = pd.DataFrame(index=faults, data=core_hours, columns=PROCESS_TYPES)
+    db = MgmtDB(f"{root_dir}/slurm_mgmt.db")
+    df = pd.DataFrame(
+        columns=proc_types,
+        index=fault_names,
+        data=np.zeros(shape=(len(fault_names), len(proc_types))),
+    )
 
+    for fault_name in fault_names:
+        rel_names = db.get_rel_names()
```

Comment:
get_rel_names should always have the same return value for no arguments? So can be outside the loop?